src/Adapter/Module/PrestaTrust/PrestaTrustChecker.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/*
+ * 2007-2017 PrestaShop
+ * 
+ * NOTICE OF LICENSE
+ * 
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ * 
+ * DISCLAIMER
+ * 
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ * 
+ *  @author PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2017 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Module\PrestaTrust;
+
+use Doctrine\Common\Cache\Cache;
+use PrestaShop\PrestaShop\Adapter\Module\Module;
+use Symfony\Component\Finder\Finder;
+
+class PrestaTrustChecker
+{
+    protected $checked_extensions = array('php', 'js', 'css', 'tpl');
+
+    const CHECKS_ALL_OK = 'Module authenticated.';
+    const CHECKS_INTEGRITY_NOK = 'Warning, the module has been modified since its purchase from the Marketplace';
+    const CHECKS_PROPERTY_NOK = 'Warning, the purchase proof is not found. This license has already been used on another shop.';
+    const CHECKS_ALL_NOK = 'Warning, the module has been modified and its purchase proof is not found.';
```

Comment:
Seems to me that it _was_ found, but that it is invalid because the module has been modified since its purchase, and thus cannot be certified to be safe anymore.

Reply:
Thanks @xBorderie! Your review will be useful for the definitive version of the wording. :)

Reply:
Is it displayed to the user or not? If yes, is this content translatable? :D

Reply:
@xBorderie : The property check (= validity of the purchase proof) and the integrity check (= module modified or not) are independent.